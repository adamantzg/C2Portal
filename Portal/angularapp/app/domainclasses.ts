
export class Product {
    id: number;
    code: string;
    name: string;
    description: string;
    combined_name: string;
}

export class Customer {
    code: string;
    name: string;
    address1: string;
    address2: string;
    address3: string;
    address4: string;
    address5: string;
    address6: string;
    town_city: string;
    county: string;
    analysis_codes1: string;
    currency: string;
    combined_name: string;
}

export class User {
    id: number;
    name: string;
    lastname: string;
    username: string;
    password: string;
    email: string;
    customer_code: string;
    address: string;
    phone: string;
    token: string;
    customer: Customer;
    roles: Role[];
    locked: boolean;
    isTopAdmin: boolean;
    isBranchAdmin: boolean;
    activated: boolean;
    isInternal: boolean;
    lastLogin: Date;
    role: string;
    role_id: number;
    permissions: Permission[];

    constructor() {
        this.customer = new Customer();
    }
}

export class Role {
    id: number;
    name: string;
    permissions: Permission[];
    // strPermissions: string[];

    constructor(id: number, name?: string, permissions?: Permission[]) {
        this.id = id;
        this.name = name;
        this.permissions = permissions;
    }
}


export class Invoice {
    customer: string;
    dated: Date | null;
    kind: string;
    reference: string;
    item_no: string;
    due_date: Date | null;
    amount: number | null;
    open_indicator: string;
    customer_order_num: string;
    selected: boolean;
    postcode: string;
}

export enum RoleId {
    User = 1,
    BranchAdmin = 2,
    Admin = 3
}

export class Order {
    order_no: string; // Sage order number
    customer_order_no: string; // Customer Order number
    date_entered: Date; // Date order received and input to Sage
    invoice_date: Date; // Currently will be to indicate "Delivered" flag. When there is an invoice date then the order is delivered
    date_required: Date;
    address1: string;
    address2: string;
    address3: string;
    address4: string;
    address5: string;
    address6: string;
    town_city: string;
    statusText: string;
    status: number;
    customer?: string;
    country: string;
    audits: OrderAudit[];
    productStockData: StockData[];
    details: OrderDetail[];
    planned: boolean;
}

export class OrderDetail {
    product: string;
    order_qty: number;
    val: number;
    order_line_no: number;
}

export  enum OrderStatus {
    Received = 5,
    InProgress = 7,
    Delivered = 8,
    Cancelled = 9
    // Processed = 'Processed',
    // Confirmed = 'Confirmed',
    // Delivered = 'Delivered',
    // Cancelled = 'Cancelled',
}
export class OrderAudit {
    audit_key: string; // need to join to the Order_no field of opheadm
    character_value: string; // limited to 6 to infer Despatch Note printed
    audit_date: Date; // is the date used for Order Processed date.
}

export class StockData {
    quantity: number;
    product: Product;
    ship_date: Date | null;
    discontinued: boolean;
    madeToOrder: boolean;
    expiresThisYear: boolean;
}

export class Holiday {
    name: string;
    date: Date;
}
export class HolidayExtracted {
    name: string;
    year: number;
    month: number;
    date: number;
    fullDate: Date;
}
export class Permission {
    id: number;
    name: string;
    selected: boolean;

    constructor(id?: number, name?: string, selected?: boolean) {
        this.id = id;
        this.name = name;
        this.selected = selected;
    }
}

export enum PermissionId {
    ViewAccountDetails = 1,
    ViewHolidayAdministration = 2,
    ViewInvoiceHistory = 3,
    ViewOrderHistory = 4,
    ViewStockSearch = 5,
    ViewUserAdministration = 6,
    ViewRoleAdministration = 7
}

export class ClearanceStock {
    product_code: string;
    long_description: string;
    range: string;
    exvat_price: number | null;
    price: number | null;
    freestock: number | null;
    discount: number;
}
