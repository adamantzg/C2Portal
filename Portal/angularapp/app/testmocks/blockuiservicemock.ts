import { EventEmitter } from '@angular/core';

export class BlockUIServiceMock {
    blockTicks = 0;
    unBlockTicks = 0;
    blockUIEvent: any = new EventEmitter<any>();

    startBlock() {
      this.blockTicks++;
    }

    stopBlock() {
      this.unBlockTicks++;
    }
}
