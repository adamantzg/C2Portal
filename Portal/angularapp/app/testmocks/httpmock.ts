import { of } from 'rxjs';

export class HttpMock {
    url: string;
    data: any;
    params: any;
    options: any;
    postResult: any;
    getResult: any;


    post<T>(url, data, params) {
      this.data = data;
      this.url = url;
      this.params = params;
      return of(this.postResult);
    }

    get<T>(url, options) {
        this.url = url;
        this.options = options;
        return of(this.getResult);
    }

  }
