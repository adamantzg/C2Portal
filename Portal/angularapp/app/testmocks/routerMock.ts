import { Component } from '@angular/core';

export class RouterMock {
    navigateUrls: string[];
    params: any;

    navigate(urls: string[], params?: any) {
        this.navigateUrls = urls;
        this.params = params;
    }
}

export class ActivatedRouteMock {

}


export class RouterLinkMock {

}

@Component({
    template: ''
  })
  export class DummyComponent {
  }
