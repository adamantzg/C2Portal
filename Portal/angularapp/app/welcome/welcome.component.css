:host{
    display: flex;
    height: 95%;
    width: 100%;
background-attachment: fixed;
background-position: center 0%;
overflow: hidden;
}
.row{
    margin-top:15px;
}
.col-sm-6 {
    text-align: center;
}
.col-sm-4 {
    margin-right: -15px
}
.col-sm-6 img{
    width: 360px;
}
.img-responsive {
    float:left;
    margin: 0px;
}
.title-holder {
    /* position: absolute; */
    float:left;
    background: rgba(255, 255, 255, .6);
    width: 100%;
    padding: 5px 0;
    margin: -155px 0;
    text-align: center;
    font-size: 28px;
    
}
.links-holder{
    /* position: absolute; */
    color:white;
    /* background: rgba(85, 172, 229, .95); */
background: rgba(53, 65, 81, .95);
    width: 100%;
    padding: 5px;
    margin: -95px 0;
    text-align: center;
    height: 80px;
    font-size: 14px;
    float:left;
    opacity: 0.7;
    /* cursor: pointer; */
}
.links-holder a{
    color:white!important;
    display: block;
    width: 100%;
    text-decoration: none;
}
.links-holder a:hover{
    opacity: .8;
    cursor:pointer;
    color:white!important;
    text-decoration: underline;
}
.glyphicon{
    font-size:12px;
    margin-right: 2px;
}
.coming-soon img {
    position: absolute;
    width: 91%;
    margin: 20px 0;
    opacity: .45;
}