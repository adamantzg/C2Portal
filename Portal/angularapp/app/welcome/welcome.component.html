<div class="container">
    <div class="row">
        <div class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2">
            <img src="assets/my_account_portal.jpg" class="img-responsive" alt="Image new picture">
            <div class="title-holder">My Account</div>
            <div class="links-holder">
                <a [routerLink]="['account']" *ngIf="userHasPermission(permissions.ViewAccountDetails)"> 
                    Account Details</a>
                <a [routerLink]="['invoice-history']" *ngIf="userHasPermission(permissions.ViewInvoiceHistory)">
                    Invoice History</a>
                <a [routerLink]="['order-history']" *ngIf="userHasPermission(permissions.ViewOrderHistory)">
                    Order History</a>
                <!-- <a [routerLink]="['order-tracking']">
                    Order History</a> -->
            </div>
        </div>
        <div class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2">
            <img src="assets/stock.jpg" class="img-responsive" alt="Image">            
            <div class="title-holder">Stock</div>
            <div class="links-holder">
                <a [routerLink]="['stock']" *ngIf="userHasPermission(permissions.ViewStockSearch)">                
                    Stock availability
                </a>
                <a [routerLink]="['clearance']" *ngIf="userHasPermission(permissions.ViewStockSearch)">                
                    Clearance stock 
                </a>    
            </div>
            
        </div>
    </div>
    <div class="row">
        <div class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2">
            <div class="coming-soon" style="display:none">
                <img src="assets/coming_soon _stamp.png">
            </div>
            <img src="assets/website.jpg" class="img-responsive" alt="Image">
            <div class="title-holder">Web site</div> 
            <div class="links-holder">
                <a class="external-link" href="http://c2portal.adamantzg.com/" target="_blank">
                    Home </a>
                <!-- <a>
                    Link 2</a>
                <a>
                    Videos</a> -->
            </div>
        </div>
        <div class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2">
            <div class="coming-soon"><img src="assets/coming_soon _stamp.png"></div>
            <img src="assets/Price_list_portal.jpg" class="img-responsive" alt="Image">
            <div class="title-holder">Price Lists </div>
            <div class="links-holder">
                <a>
                    Link 1</a>
                <a>
                    Sale</a>
                
            </div>
        </div>
    </div>
</div>
<!-- <c2-flex-body></c2-flex-body> -->