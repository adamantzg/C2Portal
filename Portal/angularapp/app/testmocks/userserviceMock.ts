import { User } from '../domainclasses';

export class UserServiceMock {
    CurrentUser: User;
}
