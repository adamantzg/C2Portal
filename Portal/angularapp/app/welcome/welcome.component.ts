import { Component, OnInit } from '@angular/core';
import { User, PermissionId } from '../domainclasses';
import { UserService } from '../services/user.service';

@Component({
  selector: 'app-welcome',
  templateUrl: './welcome.component.html',
  styleUrls: ['./welcome.component.css']
})
export class WelcomeComponent implements OnInit {

  constructor(private userService: UserService) {

   }

   user: User = this.userService.CurrentUser;
   permissions = Object.assign({}, PermissionId);

  ngOnInit() {
  }
  test() {
    alert();
    console.log('test');
  }
/***new style text  ** */
  userHasPermission(p: PermissionId) {
    return this.user.permissions.find(x => x.id === p);
  }
}
