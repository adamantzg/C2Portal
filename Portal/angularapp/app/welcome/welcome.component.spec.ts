import { TestBed, ComponentFixture } from '@angular/core/testing';
import { WelcomeComponent } from './welcome.component';
import { FormsModule } from '@angular/forms';
import { UserService } from '../services/user.service';
import { HttpClient } from '@angular/common/http';
import { HttpMock } from '../testmocks/httpmock';
import { User, PermissionId, Permission } from '../domainclasses';
import { UserServiceMock } from '../testmocks/userserviceMock';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';
import { DummyComponent } from '../testmocks/routerMock';


describe('welcome component', () => {
    let component: WelcomeComponent;
    let fixture: ComponentFixture<WelcomeComponent>;
    let userService: UserServiceMock;

    setupTestBed({
        declarations: [WelcomeComponent, DummyComponent],
        providers: [UserService, {provide: HttpClient, useClass: HttpMock}, {provide: UserService, useClass: UserServiceMock}],
        imports: [FormsModule,
            RouterTestingModule.withRoutes( [
                { path: 'account', component: DummyComponent },
                { path: 'invoice-history', component: DummyComponent },
                { path: 'order-history', component: DummyComponent },
                { path: 'stock', component: DummyComponent }
            ])]
    });

    beforeEach(() => {

        userService = TestBed.get(UserService);
        userService.CurrentUser = new User();
        userService.CurrentUser.name = 'test';
        userService.CurrentUser.customer_code = 'cust';
        fixture = TestBed.createComponent(WelcomeComponent);
        component = fixture.componentInstance;
    });

    test('it should create component', () => {
        expect(component).toBeTruthy();
        expect(fixture).toMatchSnapshot();
    });

    test('links', () => {
        component.user.permissions = [new Permission(PermissionId.ViewAccountDetails),
        new Permission(PermissionId.ViewOrderHistory)];
        fixture.detectChanges();
        let elem = fixture.debugElement.nativeElement.querySelector('a[href*="account"]');
        expect(elem).toBeTruthy();
        elem = fixture.debugElement.nativeElement.querySelector('a[href*="invoice-history"]');
        expect(elem).toBeNull();
        elem = fixture.debugElement.nativeElement.querySelector('a[href*="order-history"]');
        expect(elem).toBeTruthy();
        elem = fixture.debugElement.nativeElement.querySelector('a[href*="stock"]');
        expect(elem).toBeNull();
    });
});
