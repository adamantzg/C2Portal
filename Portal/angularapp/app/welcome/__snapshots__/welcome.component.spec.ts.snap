// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome component it should create component 1`] = `
<app-welcome
  permissions={[Function Object]}
  user={[Function User]}
  userService={[Function UserServiceMock]}
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2"
      >
        <img
          alt="Image"
          class="img-responsive"
          src="assets/img_1.jpg"
        />
        <div
          class="title-holder"
        >
          My Account
        </div>
        <div
          class="links-holder"
        >
          
          
          
        </div>
      </div>
      <div
        class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2"
      >
        <img
          alt="Image"
          class="img-responsive"
          src="assets/img_2.jpg"
        />
        <div
          class="title-holder"
        >
          Stock
        </div>
        
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2"
      >
        <div
          class="coming-soon"
        >
          <img
            src="assets/coming_soon _stamp.png"
          />
        </div>
        <img
          alt="Image"
          class="img-responsive"
          src="assets/img_3.jpg"
        />
        <div
          class="title-holder"
        >
          Instructions
        </div>
        <div
          class="links-holder"
        >
          <a>
             Link 1
          </a>
          <a>
             Link 2
          </a>
          <a>
             Videos
          </a>
        </div>
      </div>
      <div
        class="col-sm-5 col-sm-push-1 col-md-4 col-md-push-2"
      >
        <div
          class="coming-soon"
        >
          <img
            src="assets/coming_soon _stamp.png"
          />
        </div>
        <img
          alt="Image"
          class="img-responsive"
          src="assets/img_4.jpg"
        />
        <div
          class="title-holder"
        >
          News
        </div>
        <div
          class="links-holder"
        >
          <a>
             Link 1
          </a>
          <a>
             Sale
          </a>
          <a>
             Lik 3
          </a>
        </div>
      </div>
    </div>
  </div>
</app-welcome>
`;
